src/lib/formats/tzx_cas.cpp

```diff
@@ -291,10 +297,100 @@ static int tzx_cas_handle_block( int16_t **buffer, const uint8_t *bytes, int pau
 	return size;
 }
 
+static int tsx_msx_handle_block( int16_t **buffer, const uint8_t *bytes, int pause, int data_size, int pilot, int pilot_length, int bitcfg, int bytecfg, int bit0, int bit1)
+{
+	float t_scale = 1.44f;
+	int pilot_samples = tcycles_to_samplecount(pilot,t_scale);
+	int bit0_samples = tcycles_to_samplecount(bit0,t_scale);
+	int bit1_samples = tcycles_to_samplecount(bit1,t_scale);
+	int data_index;
```

Comment:
Please scope this properly.

Reply:
Fixed

Reply:
It’s still not scoped properly.

Reply:
Marking it resolved when `data_index` isn’t scoped properly doesn’t change it.

Reply:
I’m talking about scope, not indentation.  What is the useful lifetime of this variable?

Reply:
That variable is used a little further down inside the function as an index in the block's data read loop